### PR TITLE
Remove manual `torch.no_grad` and train eval mode in LightningModule

### PR DESCRIPTION
**Pull request by aniketmaurya**
PL LightningModule automatically sets train and eval mode and `torch.no_grad` during the training and validation steps.

### CODE REVIEW

File: icevision/models/torchvision/lightning_model_adapter.py [hunk 1]
```diff
@@ -36,17 +36,16 @@ def training_step(self, batch, batch_idx):
...
-
-            self.eval()
-            raw_preds = self(xb)
-            preds = self.convert_raw_predictions(
-                batch=batch, raw_preds=raw_preds, records=records
-            )
-            self.accumulate_metrics(preds=preds)
+        self.train()
+        train_preds = self(xb, yb)
+        loss = loss_fn(train_preds, yb)
```
Comment (aniketmaurya) [inline icevision/models/torchvision/lightning_model_adapter.py:41]: I wasn't able to understand why do we calculate training prediction here?

Comment (potipot) [reply icevision/models/torchvision/lightning_model_adapter.py:41]: Maybe the name is misleading , these are in fact validation preds. These are used to calculate validation loss and metrics

**Comment by potipot** [comment]
Any idea why black formatting check might failing here?
